src/lcm/functools.py

```diff
@@ -67,46 +27,66 @@ def allow_kwargs(func):
 
     # Create new signature without positional-only arguments
     new_parameters = [
-        (
-            p.replace(kind=inspect.Parameter.POSITIONAL_OR_KEYWORD)
-            if p.kind == inspect.Parameter.POSITIONAL_ONLY
-            else p
-        )
-        for p in parameters.values()
+        p.replace(kind=inspect.Parameter.KEYWORD_ONLY) for p in parameters.values()
     ]
     new_signature = signature.replace(parameters=new_parameters)
 
     @functools.wraps(func)
-    def allow_kwargs_wrapper(*args, **kwargs):
+    def func_with_only_kwargs(*args, **kwargs):
+        if args:
+            raise ValueError(
+                (
+                    "This function was decorated with allow_only_kwargs, but was "
```

Comment:
```suggestion
                    "This function has been decorated so that it allows only kwargs, but was "
```
Decorator has been renamed, better use language.